geode-core/src/main/java/org/apache/geode/internal/cache/DistTXPrecommitMessage.java

```diff
@@ -315,7 +315,7 @@ public void waitForPrecommitCompletion() {
     }
 
     @Override
-    protected void processException(DistributionMessage msg, ReplyException ex) {
+    protected synchronized void processException(DistributionMessage msg, ReplyException ex) {
```

Comment:
We may not need synchronized(this) in this method.

Reply:
The parents are synchronized and the override is not. This is not considered a good practice. I will look into this a bit more.